lib/addressable/uri.rb

```diff
@@ -1032,9 +1032,9 @@ def normalized_userinfo
         if !current_user && !current_password
           nil
         elsif current_user && current_password
-          "#{current_user}:#{current_password}"
+          "#{current_user}:#{current_password}".dup
         elsif current_user && !current_password
-          "#{current_user}"
+          "#{current_user}".dup
```

Comment:
Yeah, this line looks odd to me. I probably wrote it that way absent-mindedly, but this code just seems _bad_ to me looking back on it.